docs/_static/searchtools.js

```diff
@@ -496,19 +519,41 @@ var Search = {
           break;
         }
       }
-
       // if we have still a valid result we can add it to the result list
       if (valid) {
+        /**
+         * file: index
+         * docnames: array of paths
+         * titles: array of titles
+         */
+
         // select one (max) score for the file.
         // for better ranking, we should calculate ranking by using words statistics like basic tf-idf...
         var score = $u.max($u.map(fileMap[file], function(w){return scoreMap[file][w]}));
-        function getParentTitle(f) {
+
+        function get1stLevelAncestor(f) {
```

Comment:
I would prefer to spell out the word

```suggestion
        function getFirstLevelAncestor(f) {
```

...or...

```suggestion
        function getLevel1Ancestor(f) {
```

...because in Spanish, the ordinal number is either 1o or 1a, depending on the gender of "Level".

Either refactoring is acceptable.

Reply:
replaced by function that gets not only title but array of titles of ancestors.